java/arcs/core/data/util/ReferencablePrimitive.kt

```diff
@@ -31,8 +31,9 @@ data class ReferencablePrimitive<T>(
     val valueRepr: String = value.toString()
```

Comment:
can we also not store valueRepr (remove 'val' here), instead use value.toString as part of id.get()?

Reply:
I thought about its tradeoff between memory footprint and performance.
Then decided using few bytes to expedite its toString() serialization and reduce times of traversal of each field to generate filed's string representation while querying id.